ui/app/models/azure/config.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Model, { attr } from '@ember-data/model';
+import fieldToAttrs, { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+
+// Note: while the API docs indicate subscriptionId and tenantId are required, the UI does not enforce this because the user may pass these values in as environment variables.
+// https://developer.hashicorp.com/vault/api-docs/secret/azure#configure-access
+export default class AzureConfig extends Model {
+  @attr('string') backend; // dynamic path of secret -- set on response from value passed to queryRecord
+  @attr('string', {
+    label: 'Subscription ID',
+  })
+  subscriptionId;
+  @attr('string', {
+    label: 'Tenant ID',
+  })
+  tenantId;
+  @attr('string', {
+    label: 'Client ID',
+  })
+  clientId;
```

Comment:
nit: totally up to you, but i find these easier to read if they're on a single line when the options object is so small 
```suggestion
  @attr('string', { label: 'Subscription ID' }) subscriptionId;
  @attr('string', { label: 'Tenant ID' }) tenantId;
  @attr('string', { label: 'Client ID' }) clientId;
```